### ISSUE TITLE
It has become more difficult to import PrismaClient in Deno 2.1

### ISSUE THREAD

**Issue by kawmra** [denoland/deno#27079]
Version: Deno 2.1.1

Hello,

First of all, I’m really excited about the improvements in Deno 2.1.

However, since Deno 2.1 fixed the bug where files required from .cjs files were automatically treated as CommonJS, I've been unable to import Prisma Client.

I understand that this is not an issue with Deno itself, but rather one of the many compatibility issues with Prisma.
Nevertheless, as it relates to Deno's compatibility with CommonJS, I'd like to ask a question here.

Before Deno 2.1, we used a helper script to import Prisma Client:

```js
// prisma-client.cjs
const mod = require("./generated/client/index.js");

Object.keys(mod).forEach((key) => {
    exports[key] = mod[key];
});
```

```ts
// prisma-client.ts
// @deno-types="./generated/client/deno/index.d.ts"
export * from "./prisma-client.cjs";
```

This allowed imports in other script files like:

```ts
// foo.ts
import { PrismaClient } from "./prisma-client.ts";
```

This approach depended on the behavior fixed in Deno 2.1, where files required from .cjs were automatically considered CommonJS.

When updating to Deno 2.1, I expected that this helper script would become unnecessary and that we could import like this:

```ts
// package.json
{
    "type": "commonjs",
    // other settings here...
}
```

```ts
// foo.ts
import { PrismaClient } from "./generated/client/index.js";
```

Unfortunately, this didn't work as expected because Prisma generates its own package.json file in `./generated/client/package.json`, which doesn't include "type": "commonjs".

Since there are several situations Prisma Client is regenerated, modifying `./generated/client/package.json` or changing filenames to `.cjs` after each regeneration isn't a sustainable solution.

I’d really appreciate any advice or suggestions on how to fix this situation.

Thank you!

**Comment by unimonkiez** [comment]
I tried to use your method and also change the generated `package.json` to have ` "type": "commonjs"` (Just so I'll have a solution in the meantime), but still no luck:
```bash
> deno run --watch --allow-import --allow-read --allow-env --allow-net --allow-ffi src/start.ts
Watcher Process started.
error: Cannot require ES module 'file:///Users/yuvalsaraf/dev/manager/app/graphql-device/src/prisma-generated/index.js' from 'file:///Users/yuvalsaraf/dev/manager/app/graphql-device/src/prisma-client.cjs'
```

**Comment by kawmra** [comment]
@unimonkiez Adding `"type": "commonjs"` to `./generated/client/package.json` works for me with Deno 2.1.1.
In this case, you should import `./generated/client/index.js` directly instead of wrapping with `prisma-client.cjs` I think.

**Comment by unimonkiez** [comment]
@kawmra 
I did reproduce the bug [here](https://github.com/unimonkiez/deno-commonjs-reproduce), I think it's a bigger issue than the one mentioned here.
Unless I do something wrong.
I did include type: commonjs but still get the error
```
error: Uncaught (in promise) ReferenceError: exports is not defined
Object.defineProperty(exports, "__esModule", { value: true });
                      ^
    at file:///Users/yuvalsaraf/dev/deno-commonjs/src/generated-code/index.js:1:23

    info: Deno supports CommonJS modules in .cjs files, or when the closest
          package.json has a "type": "commonjs" option.
    hint: Rewrite this module to ESM,
          or change the file extension to .cjs,
          or add package.json next to the file with "type": "commonjs" option.
    docs: https://docs.deno.com/go/commonjs
Watcher Process failed. Restarting on file change...
```

**Comment by unimonkiez** [comment]
@dsherret I saw you're trying to fix this with #27094 , but ultimately prisma with need to generated a `package.json` with `type: commonjs`, but that also doesn't seem to work with [my reproduction](https://github.com/unimonkiez/deno-commonjs-reproduce).
Can you check it out?
it will provide a much nicer solution than the flag.

**Comment by kawmra** [comment]
@unimonkiez I was able to reproduce the behavior using your repro code.
In my repository, there was a package.json at the root, and when I added an empty package.json (`{}`) to your root dir, it started working. I'm not sure the reason.

**Comment by bartlomieju** [comment]
The change in Prisma has landed and it's supposed to be released in Prisma v6.

**Comment by kawmra** [comment]
Thanks for your support!

**Comment by bartlomieju** [comment]
Can we close this issue with https://github.com/prisma/prisma/releases/tag/6.0.0 released?

**Comment by kawmra** [comment]
It worked as expected! Thanks again!!

**Comment by kawmra** [comment]
@bartlomieju Sadly, it looks like https://github.com/prisma/prisma/pull/25734 was reverted in [v6.0.1](https://github.com/prisma/prisma/releases/tag/6.0.1) 😢

**Comment by mediaformat** [comment]
> `deno run --watch --allow-import --allow-read --allow-env --allow-net --allow-ffi src/start.ts`

Since prisma generates commonjs files, what worked for me was adding the --unstable-detect-cjs feature flag:
`deno run --watch --allow-import --allow-read --allow-env --allow-net --allow-ffi --unstable-detect-cjs src/start.ts`

**Comment by dsherret** [comment]
Yeah, the workaround here is to add the following to a deno.json:

```json
{
  "unstable": ["detect-cjs"]
}
```

Or run with `--unstable-detect-cjs`. That will cause Deno to run a bit slower and detect whether a module is CJS when there's no explicit `{ "type": "commonjs" }`.

The real fix will be for next.js to fix this https://github.com/vercel/next.js/issues/35110#issuecomment-2016567633 and then for prisma to add `{ "type": "commonjs" }` in the output, which will also make it compatible with future versions of Node.js that may switch to a default `{ "type": "module" }` out of the box.
